File: Baekjoon/BarkingDog/10866 - 덱.cpp
#include<iostream>
#include<deque>
#define endl "\n"

using namespace std;

int main()
{
	ios::sync_with_stdio, cin.tie(0);

	deque<int> dq;

	int n, input;
	string order;
	
	cin >> n;
	
	for (int i = 0; i < n; i++)
	{
		cin >> order;
		if (order == "push_front")
		{
			cin >> input;
			dq.push_front(input);
		}
		else if (order == "push_back")
		{
			cin >> input;
			dq.push_back(input);
		}
		else if (order == "pop_front")
			if (dq.empty())	cout << "-1" << endl;
			else {
				cout << dq.front() << endl;
				dq.pop_front();
			}
		else if (order == "pop_back")
			if (dq.empty())	cout << "-1" << endl;
			else {
				cout << dq.back() << endl;
				dq.pop_back();
			}
		else if (order == "size")
			cout << dq.size() << endl;
		else if (order == "empty")
			if (dq.empty())	cout << "1" << endl;
			else cout << "0" << endl;
		else if (order == "front")
			if (dq.empty())	cout << "-1" <<endl;
			else cout << dq.front() << endl;
		else if (order == "back")
			if (dq.empty())	cout << "-1" << endl;
			else cout << dq.back() << endl;
		
	}
	system("pause");
}

File: Baekjoon/10866 - 덱.java
import java.io.IOException;
import java.util.Deque;
import java.io.BufferedReader;
import java.util.StringTokenizer;
import java.io.InputStreamReader;
import java.util.LinkedList;

public class Main {
	public static void main(String args[]) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringBuilder sb = new StringBuilder();

		int n = Integer.parseInt(br.readLine());

		StringTokenizer st;

		Deque<Integer> dq = new LinkedList<>();

		while (n-- > 0) {
			st = new StringTokenizer(br.readLine(), " ");
			switch (st.nextToken()) {
			case "push_front":
				dq.addFirst(Integer.parseInt(st.nextToken()));
				break;
			case "push_back":
				dq.addLast(Integer.parseInt(st.nextToken()));
				break;
			case "pop_front":
				if (dq.isEmpty())
					sb.append("-1\n");
				else
					sb.append(dq.removeFirst() + "\n");
				break;
			case "pop_back":
				if (dq.isEmpty())
					sb.append("-1\n");
				else
					sb.append(dq.removeLast() + "\n");
				break;
			case "size":
				sb.append(dq.size() + "\n");
				break;
			case "empty":
				if (dq.isEmpty())
					sb.append("1\n");
				else
					sb.append("0\n");
				break;
			case "front":
				if (dq.isEmpty())
					sb.append("-1\n");
				else
					sb.append(dq.peekFirst() + "\n");
				break;
			case "back":
				if (dq.isEmpty())
					sb.append("-1\n");
				else
					sb.append(dq.peekLast() + "\n");
				break;

			}
		}
		System.out.println(sb);
	}
}